tests/baseline/cases/errata-containers-legacy-repos.yml

```diff
@@ -1779,6 +1779,16 @@ items:
 - ErratumPushItem:
     build: null
     build_info: null
+    container_list:
+    - openshift4/ose-local-storage-rhel7-operator-metadata:
+        digest: sha256:2321a7d13d9fa53f05437663cf2dc217d15f3cda4b67076c941b10f0491cf9d7
+        images:
+          docker-image-sha256:05649a19718fde131372c761d359302ccbe81f9744d2893ac4c826b05d670206.ppc64le.tar.gz:
```

Comment:
I'm questioning whether it really makes sense to expose the koji archive names here.

I know that's what ET is doing, but you already wrote code to replace the koji *build* NVR (like local-storage-operator-metadata-container-v4.3.28.202006290519.p0.prod-1) with the more useful and generic repo names (like openshift4/ose-local-storage-rhel7-operator-metadata).  So you already set the precedent that we'll patch up the data provided by ET into something more useful for the use-case.

In that case, does it make more sense to expose the arch here instead?  I think the users of this data wanted to know the arch of each image, didn't they?  And right now they'd probably be forced to write some code like: split the key on "." and the third-last element is the image's arch.

Maybe we should take care of that for them?  Right now the format seems like a quirk of how ET and koji work together, and we can take care of it right here where we query ET since we know how ET works.  We can then expose the data as one of these formats:

```yaml
images:
- arch: s390x
  digest: (...the digest)
- arch: x86_64
  digest: (...the digest)
```

...or:

```yaml
images:
  s390x:
    digest: (...the digest)
  x86_64:
    digest: (...the digest)
```

Either one I suppose is more useful for the code wanting to access this data.

It would also be easier to ensure that data remains compatible over time, seeing as there is no guarantee that container images are ever stored into separate files per arch like this - that's just a detail of how it currently works in koji.

Reply:
I was also thinking the same way. But same as the format of storing images can change, so does the archive name. And as pub is only the middle man here I didn't want  to involve here more transformations than necessary.